pkg/controller/linstorcontrollerset/linstorcontrollerset_controller.go

```diff
@@ -207,13 +189,13 @@ func (r *ReconcileLinstorControllerSet) Reconcile(request reconcile.Request) (re
 
 	markedForDeletion := pcs.GetDeletionTimestamp() != nil
 	if markedForDeletion {
-		finished, err := r.finalizeControllerSet(pcs)
+		finished, err := r.finalizeControllerSet(ctx, pcs)
```

Comment:
If we're going to go with this `TemporaryError` approach, we should adjust this method to use that too.